posts/2023/06/29/how-to-use-the-new-constructed-inventory-feature-in-aap-2.4.md

```diff
@@ -0,0 +1,178 @@
+
+The jinja2 template to create “**target_group**” will include a host if, when inspecting that host, the hostvar of “**account_alias**” exists **and** is equal to “**product_dev**”. 
+
+The “**| default**” syntax is necessary in the event that the variable is not defined for some hosts. 
+
+The “**strict: true**” dictates that referencing an undefined variable will fail the inventory update, making the “| default” necessary. 
+
+Using “|default| and the strict parameter is best practice, in order to force you to make the undefined case explicit.
+
+Constructing groups can be useful to add groups on-the-fly to be referenced by playbooks. Why would you want to? Sometimes it is convenient to synthesize groups from hostvars, because you can do things with groups that you can’t do with hostvars, like using in host patterns, like with **limit** in this example. 
+
+If you look at all the produced hosts for our constructed inventory:
+![inventory group host](/images/posts/how-to-use-the-new-constructed-inventory-feature-in-aap-2.4/inventory-group-host.png)
+
+We can see in this example that **host3** in the **east** inventory and **host5** in the **west** inventory are not included. This is because in the input inventories, they are in a different account (account_4321) which has an “account_alias” that does not match our specified “product_dev” value. Note that even though the group for account_4321 was imported into the constructed inventory, no hosts in that group matched our requirement so the imported group is empty in our constructed inventory.
+
+Source vars input can include **host** variables, in addition to adding groups.
+
+Alan’s Github repository also contains another example, which you may find useful. In [construct.yml](https://github.com/AlanCoding/Ansible-inventory-file-examples/blob/master/issues/AWX371/construct.yml) we resolve the ‘state’ of a host, and assign it to a number of groups based on whether its shutdown or part of a particular environment (dev). The source of truth for this .yml file can come from other systems outside of Ansible Automation Platform, and we can use it to perform automation runs against subsets of hosts on-the-fly.
+
+# Debugging Tips
+To use the prior example, when developing the constructed inventory, consider that you delete the limit value and change the source-vars to the following content.
+
+![constructed groups details](/images/posts/how-to-use-the-new-constructed-inventory-feature-in-aap-2.4/inventory-group--combine-details.webp)
+
+This will run a similar template of and save it to a variable named “effective_account_alias”. By making *limit* blank, we will be sure to get all hosts from the input inventories. This would allow us to inspect fine-grained details of the inclusion criteria on individual hosts, shown below for host3.
```

Comment:
```suggestion
This will run a similar template of and save it to a variable named “effective_account_alias”. By making **limit** blank, we will be sure to get all hosts from the input inventories. This would allow us to inspect fine-grained details of the inclusion criteria on individual hosts, shown below for host3.
```